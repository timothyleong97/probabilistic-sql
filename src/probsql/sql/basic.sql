CREATE TABLE test(id SERIAL, gate GATE);
INSERT INTO test(gate) VALUES ('gaussian(1.0, 2.0)'), ('poisson(3.0)');
SELECT * FROM test;
SELECT 'gaussian(1.0, 2.0)'::gate + 'poisson(3.0)'::gate AS comp_var;
SELECT 'gaussian(1.0, 2.0)'::gate - 'poisson(3.0)'::gate AS comp_var;
SELECT 'gaussian(1.0, 2.0)'::gate * 'poisson(3.0)'::gate AS comp_var;
SELECT 'gaussian(1.0, 2.0)'::gate / 'poisson(3.0)'::gate AS comp_var;
