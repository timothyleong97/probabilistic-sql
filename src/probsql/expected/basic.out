CREATE TABLE test(id SERIAL, gate GATE);
INSERT INTO test(gate) VALUES ('gaussian(1.0, 2.0)'), ('poisson(3.0)');
SELECT * FROM test;
 id |         gate         
----+----------------------
  1 | gaussian(1.00, 2.00)
  2 | poisson(3.00)
(2 rows)

SELECT 'gaussian(1.0, 2.0)'::gate + 'poisson(3.0)'::gate AS comp_var;
                comp_var                
----------------------------------------
 (gaussian(1.00, 2.00))+(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate - 'poisson(3.0)'::gate AS comp_var;
                comp_var                
----------------------------------------
 (gaussian(1.00, 2.00))-(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate * 'poisson(3.0)'::gate AS comp_var;
                comp_var                
----------------------------------------
 (gaussian(1.00, 2.00))*(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate / 'poisson(3.0)'::gate AS comp_var;
                comp_var                
----------------------------------------
 (gaussian(1.00, 2.00))/(poisson(3.00))
(1 row)

SELECT 2::gate AS const_var;
      const_var       
----------------------
 gaussian(2.00, 0.00)
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate < 'poisson(3.0)'::gate AS my_cond;
                my_cond                 
----------------------------------------
 (gaussian(1.00, 2.00))<(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate <= 'poisson(3.0)'::gate AS my_cond;
                 my_cond                 
-----------------------------------------
 (gaussian(1.00, 2.00))<=(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate <> 'poisson(3.0)'::gate AS my_cond;
                 my_cond                 
-----------------------------------------
 (gaussian(1.00, 2.00))!=(poisson(3.00))
(1 row)

SELECT 'gaussian(1.0, 2.0)'::gate <> 2 AS my_cond;
                    my_cond                     
------------------------------------------------
 (gaussian(1.00, 2.00))!=(gaussian(2.00, 0.00))
(1 row)

